xformers/factory/block_factory.py

```diff
@@ -135,7 +137,10 @@ def __init__(
         )
 
         # Default is that this layer is the only one, so both first and last
-        self.layer_position = LayerPosition()
+        if layer_position:
```

Comment:
Thanks a lot for catching this bug @jajsmith! Perhaps we can add a test for this [here](https://github.com/facebookresearch/xformers/blob/main/tests/test_model_factory.py#L110)?

Just cycle through self.encoders and self.decoders and assert that the layer_position is correct?